active/teachingtip/TeachingTipSpec.md

```diff
@@ -0,0 +1,690 @@
+
+When the Save button is added to the visual tree, a teaching tip will appear and display a message to the user. 
+
+XAML
+```XAML
+<Button x:Name="SaveButton" Content="Save" Loaded="SaveButton_Loaded">
+    <Button.Resources>
+        <muxc:TeachingTip x:Name="AutoSaveTip"
+            Target="{x:Bind SaveButton}"
+            Title="Save automatically"
+            Subtitle="When you save your file to OneDrive, we save your changes as you go - so you never have to.">
+        </muxc:TeachingTip>
+    </Button.Resources>
+</Button>
+```
+
+C#
+```C#
+public MainPage()
+{
+    this.InitializeComponent();
+
+    if(!HaveExplainedAutoSave())
+    {
+        AutoSaveTip.IsOpen = true;
+        SetHaveExplainedAutoSave();
+    }
+}
+```
+
+Here's the result when this code runs and the Save button gets loaded.
+
+![A sample app with a teaching tip pointing at the save button. The tip title reads "Saving automatically" and the subtitle reads "We save your changes as you go - so you never have to." There is a close button on the top right corner of the teaching tip.](TeachingTipPointerSample.jpg)
+
+The IsOpen property also enables teaching tip to be implemented with data binding.
+
+XAML
+```XAML
+<Button x:Name="SaveButton" Content="Save" Loaded="SaveButton_Loaded">
+    <Button.Resources>
+        <muxc:TeachingTip x:Name="AutoSaveTip"
+            IsOpen="{x:Bind ShowAutoSaveTip, Mode=TwoWay}"
+            Target="{x:Bind SaveButton}"
+            Title="Saving automatically"
+            Subtitle="We save your changes as you go - so you never have to.">
+        </muxc:TeachingTip>
+    </Button.Resources>
+</Button>
+```
+
+C#
+```C#
+public bool ShowAutoSaveTip
+{
+    get { return (bool)GetValue(ShowAutoSaveTipProperty); }
+    set { SetValue(ShowAutoSaveTipProperty, value); }
+}
+ 
+public static readonly DependencyProperty ShowAutoSaveTipProperty = DependencyProperty.Register("ShowAutoSaveTip", typeof(bool), typeof(MainPage), new PropertyMetadata(false));
+
+private void SaveButton_Loaded(object sender, RoutedEventArgs e)
+{
+    AutoSaveTip.IsOpen = true;
+}
+```
+
+### Non-pointing tips
+
+Not all tips relate to an element onscreen. For these scenarios, do not set the Target property and the teaching tip will instead display relative to the edges of the XamlRoot. However, a teaching tip can have the pointer removed while retaining placement relative to a UI element by setting the PointerMode property to "Off". 
+
+Unlike pointing teaching tips which can only be added to a resource dictionary, non-pointing teaching tips can also be added through the visual tree, as shown below.
+
+XAML
+```XAML
+<Button x:Name="SaveButton" Content="Save" Click="OnFirstSaveButtonClick">
+
+<muxc:TeachingTip x:Name="AutoSaveTip"
+    Title="Saving automatically"
+    Subtitle="We save your changes as you go - so you never have to.">
+</muxc:TeachingTip>
+```
+
+C#
```

Comment:
This C# doesn't actually contribute to the example, and it's distracting because it doesn't contain the right IsOpen logic like the first example does. Could you just remove this C# portion and just leave the markup? Same for most/all of the rest of the examples.

Reply:
@MikeHillberg The [Password Box guidance](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/password-box) I based this formatting off of labels the langauges used and I imagine this detail will be needed for transfering this documentation to MSDN? Would it be preferred to instead make these examples complete like the one above? It is still on my list to rework these examples today to make them more like the first one.

Reply:
I think lets leave the C# out for now and talk to CX about guidance for a case like this.

Reply:
Sounds good, I will make this change.

Reply:
[Committed](https://github.com/Microsoft/microsoft-ui-xaml-specs/commit/899577cbfd50a626cf95f93f421db1d1a35fc23e).